hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoCreateHandler.java

```diff
@@ -49,14 +76,200 @@ public void preHandle(@NonNull final PreHandleContext context) {
+        }
+        if (newPayerBalance < 0) {
+            throw new HandleException(INSUFFICIENT_PAYER_BALANCE);
+        }
+        // TODO: check if payer account is detached when we have started expiring accounts ?
+    }
+
+    /**
+     * Builds an account based on the transaction body and the consensus time.
+     *
+     * @param op the transaction body
+     * @param handleContext the handle context
+     * @return the account created
      */
-    public void handle() {
-        throw new UnsupportedOperationException("Not implemented");
+    private Account buildAccount(CryptoCreateTransactionBody op, HandleContext handleContext) {
+        long autoRenewPeriod = op.autoRenewPeriodOrThrow().seconds();
+        long consensusTime = handleContext.consensusNow().getEpochSecond();
+        long expiry = consensusTime + autoRenewPeriod;
+        var builder = Account.newBuilder()
+                .memo(op.memo())
+                .expiry(expiry)
+                .autoRenewSecs(autoRenewPeriod)
+                .receiverSigRequired(op.receiverSigRequired())
+                .maxAutoAssociations(op.maxAutomaticTokenAssociations())
+                .tinybarBalance(op.initialBalance())
+                .declineReward(op.declineReward());
+
+        if (onlyKeyProvided(op)) {
+            builder.key(op.keyOrThrow());
+        } else if (keyAndAliasProvided(op)) {
+            builder.key(op.keyOrThrow()).alias(op.alias());
+        }
+
+        if (op.hasStakedAccountId() || op.hasStakedNodeId()) {
+            final var stakeNumber = getStakedId(op);
+            builder.stakedNumber(stakeNumber);
+        }
+        // set the new account number
+        builder.accountNumber(handleContext.newEntityNumSupplier().getAsLong());
+        return builder.build();
+    }
+
+    /**
+     * Checks if only key is provided.
+     *
+     * @param op the transaction body
+     * @return true if only key is provided, false otherwise
+     */
+    private boolean onlyKeyProvided(@NonNull final CryptoCreateTransactionBody op) {
+        return op.hasKey() && op.alias().equals(Bytes.EMPTY);
+    }
+
+    /**
+     * Checks if both key and alias are provided.
+     *
+     * @param op the transaction body
+     * @return true if both key and alias are provided, false otherwise
+     */
+    private boolean keyAndAliasProvided(@NonNull final CryptoCreateTransactionBody op) {
+        return op.hasKey() && !op.alias().equals(Bytes.EMPTY);
+    }
+
+    /**
+     * Gets the stakedId from the provided staked_account_id or staked_node_id.
+     * When staked_node_id is provided, it is stored as negative number in state to
+     * distinguish it from staked_account_id. It will be converted back to positive number
+     * when it is retrieved from state.
+     *
+     * To distinguish for node 0, it will be stored as - node_id -1.
+     * For example, if staked_node_id is 0, it will be stored as -1 in state.
```

Comment:
This seems different than the usual approach AFAIK. Don't we usually figure out which value belongs to which property based on its position? 

I.e. something like
```java
stakedId = nextInt();
stakedAccountId = nextInt();
stakedNodeId = nextInt();
```

I guess I don't see the reason for differentiating based on +/- numbers. Seems dangerous and not very scaleable. Is this how it's done currently?

Reply:
No .. it is not while serializing. The way we store StakedID is that way. If its staked to node we store as `-nodeId-1`  and if staked to account, we store account number. We could have stored in two different variables, but this seemed to be fine.